pandas/core/frame.py

```diff
@@ -2212,6 +2212,22 @@ def insert(self, loc, column, value, allow_duplicates=False):
         self._data.insert(
             loc, column, value, allow_duplicates=allow_duplicates)
 
+    def transform(self, **kwargs):
+        """
+
+        """
+        data = self.copy()
```

Comment:
we never shallow copy
-
much more trouble than it's worth
only indexes are shallow